src/Integrations/Integrations/Symfony/PathExtractor.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace DDTrace\Integrations\Symfony;
+
+use Symfony\Component\Routing\Annotation\Route as RouteAnnotation;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionAttribute;
+use ReflectionException;
+
+class PathExtractor
+{
+    protected $routeAnnotationClass = RouteAnnotation::class;
+    protected $defaultRouteIndex = 0;
+
+
+    /**
+     * Sets the annotation class to read route properties from.
+     */
+    public function setRouteAnnotationClass($class)
+    {
+        $this->routeAnnotationClass = $class;
+    }
+
+    public function extract($classMethod, $routeName, $locale)
+    {
+        //This method is only available on PHP8. On coming PRs this class will also parse Docblocks
+        if (!\method_exists(ReflectionClass::class, 'getAttributes')) {
+            return;
+        }
+
+        $className = $classMethod; //It may not come with method when invokable controller
+        $methodName = null;
+        if (str_contains($classMethod, "::")) {
+            $exploded = explode("::", $classMethod);
+            $className = $exploded[0];
+            $methodName = $exploded[1];
+        }
+
+        if (!class_exists($className)) {
+            return;
+        }
+        $class = new ReflectionClass($className);
+        if ($class->isAbstract()) {
+            return;
+        }
```

Comment:
Why do we return if the class is abstract? Why could we even have an abstract class here? Does it mean `$request->attributes->get('_controller')` references an abstract class?

Reply:
I based this algo on `https://github.com/symfony/routing/blob/6.2/Loader/AnnotationClassLoader.php` but you're right, since we are on runtime(as opposed to that class), we should not get an abstract here. I'll remove it. Thanks

Reply:
Thanks for the link :)